failure in wallet_basic.py --descriptors

https://cirrus-ci.com/task/6282049166770176?logs=ci#L3112
```bash
 node0 2023-03-12T18:59:07.047547Z [scheduler] [wallet/wallet.h:840] [WalletLogPrintf] [default_wallet] AddToWallet 0a94aab7b4dc8ba22b88ccd6b2c9898936c7d36bf8b3e1789f6083b85f6d752a  new 
 test  2023-03-12T18:59:07.053000Z TestFramework (ERROR): Assertion failed 
                                   Traceback (most recent call last):
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-i686-pc-linux-gnu/test/functional/test_framework/test_framework.py", line 134, in main
                                       self.run_test()
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-i686-pc-linux-gnu/test/functional/wallet_basic.py", line 727, in run_test
                                       assert_equal(len(coins), 2)
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-i686-pc-linux-gnu/test/functional/test_framework/util.py", line 56, in assert_equal
                                       raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
                                   AssertionError: not(1 == 2)
 test  2023-03-12T18:59:07.054000Z TestFramework (DEBUG): Closing down network thread 
```

> https://cirrus-ci.com/task/6282049166770176?logs=ci#L3112
> 
> ```shell
>  node0 2023-03-12T18:59:07.047547Z [scheduler] [wallet/wallet.h:840] [WalletLogPrintf] [default_wallet] AddToWallet 0a94aab7b4dc8ba22b88ccd6b2c9898936c7d36bf8b3e1789f6083b85f6d752a  new 
>  test  2023-03-12T18:59:07.053000Z TestFramework (ERROR): Assertion failed 
>                                    Traceback (most recent call last):
>                                      File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-i686-pc-linux-gnu/test/functional/test_framework/test_framework.py", line 134, in main
>                                        self.run_test()
>                                      File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-i686-pc-linux-gnu/test/functional/wallet_basic.py", line 727, in run_test
>                                        assert_equal(len(coins), 2)
>                                      File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-i686-pc-linux-gnu/test/functional/test_framework/util.py", line 56, in assert_equal
>                                        raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
>                                    AssertionError: not(1 == 2)
>  test  2023-03-12T18:59:07.054000Z TestFramework (DEBUG): Closing down network thread 
> ```

name: Pull request labeler
on: [ pull_request_target ]

permissions:
  contents: read
  pull-requests: write

jobs:
  triage:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/labeler@v4
        with:
          repo-token: ${{ secrets.GITHUB_TOKEN }}

https://cirrus-ci.com/task/5029024158711808?logs=ci#L2900
```bash
 node0 2023-06-23T16:51:00.034382Z [scheduler] [wallet/sqlite.cpp:47] [TraceSqlCallback] [/tmp/cirrus-ci-build/ci/scratch/test_runner/test_runner_₿_🏃_20230623_164625/wallet_basic_242/node0/regtest/wallets/default_wallet/wallet.dat] SQLite Statement: INSERT or REPLACE into main values(?, ?) 
 test  2023-06-23T16:51:00.043000Z TestFramework (ERROR): Assertion failed 
                                   Traceback (most recent call last):
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-i686-pc-linux-gnu/test/functional/test_framework/test_framework.py", line 132, in main
                                       self.run_test()
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-i686-pc-linux-gnu/test/functional/wallet_basic.py", line 746, in run_test
                                       assert_equal(len(coins), 2)
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-i686-pc-linux-gnu/test/functional/test_framework/util.py", line 57, in assert_equal
                                       raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
                                   AssertionError: not(1 == 2)
 test  2023-06-23T16:51:00.044000Z TestFramework (DEBUG): Closing down network thread 
```

--- a/test/functional/wallet_basic.py
+++ b/test/functional/wallet_basic.py
@@ -656,6 +656,9 @@ class WalletTest(BitcoinTestFramework):
 
         # check that it works again with -spendzeroconfchange set (=default)
         self.restart_node(0, ["-spendzeroconfchange=1"])
+        # Make sure the wallet knows the tx in the mempool
+        self.nodes[0].syncwithvalidationinterfacequeue()
+
         zeroconf_wallet = self.nodes[0].get_wallet_rpc("zeroconf")
         utxos = zeroconf_wallet.listunspent(minconf=0)
         assert_equal(len(utxos), 1)

--- a/test/functional/wallet_basic.py
+++ b/test/functional/wallet_basic.py
@@ -603,6 +603,8 @@ class WalletTest(BitcoinTestFramework):
         txid_a = self.nodes[0].sendtoaddress(addr_a, 0.01)
         txid_b = self.nodes[0].sendtoaddress(addr_b, 0.01)
         self.generate(self.nodes[0], 1, sync_fun=self.no_op)
+        # Prevent race of listunspent with outstanding TxAddedToMempool notifications
+        self.nodes[0].syncwithvalidationinterfacequeue()
         # Now import the descriptors, make sure we can identify on which descriptor each coin was received.
         self.nodes[0].createwallet(wallet_name="wo", descriptors=True, disable_private_keys=True)
         wo_wallet = self.nodes[0].get_wallet_rpc("wo")

